frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -104,37 +104,51 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
     }
   }, [buildingToAssociateLand]);
 
-  /** query pims for the given pid, set data within the form if match found. Fallback to querying the parcel data layer. */
-  const handlePidChange = (pid: string, nameSpace?: string) => {
-    fetchParcelsDetail({ pid } as any)(dispatch).then(resp => {
+  /**
+   * Attempt to fetch the parcel within PIMS matching the passed pid or pin value. If that request fails, make another request to the parcel layer with the same data.
+   * @param pidOrPin
+   * @param parcelLayerSearchCallback
+   * @param nameSpace
+   */
+  const fetchPimsOrLayerParcel = (
```

Comment:
this method is now used in all areas where a pid/pin search occurs, which ensures consistency.